code/modules/mob/living/silicon/silicon_defense.dm

```diff
@@ -77,7 +77,8 @@
 			visible_message(span_notice("[user] pets [src]."), \
 							span_notice("[user] pets you."), null, null, user)
 			to_chat(user, span_notice("You pet [src]."))
-			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT_RND, "pet_borg", /datum/mood_event/pet_borg)
+			if (user.mind.assigned_role in list(JOB_RESEARCH_DIRECTOR, JOB_SCIENTIST, JOB_ROBOTICIST, JOB_GENETICIST))
```

Comment:
needing to define behavior externally like this seems like a step down. shouldnt mobs with that role start accepting that mood event and all other mobs reject it?

Reply:
When I was working on my mood refactor, I actually came to the same conclusion and implementation as Nova.  Having a separate signal for this is bloat and overkill.  It also unintentionally makes the code harder to read since a _small chunk_ of code is split into 3 different files needlessly fragmenting it.  It'd make sense if the signal was going to be used in other places, but in this case it's essentially a singleton.

This isn't the only signal to do this.  I've ran into this in other places and it always makes me shake my head since I think it's poor code practice and should be discouraged.